storage_scrubber/src/pageserver_physical_gc.rs

```diff
@@ -484,8 +598,41 @@ pub async fn pageserver_physical_gc(
     let accumulator = Arc::new(std::sync::Mutex::new(TenantRefAccumulator::default()));
 
     // Generate a stream of TenantTimelineId
-    let timelines = tenants.map_ok(|t| stream_tenant_timelines(&remote_client, &target, t));
-    let timelines = timelines.try_buffered(CONCURRENCY);
+    enum GcSummaryOrContent<T> {
+        Content(T),
+        GcSummary(GcSummary),
+    }
+    let timelines = tenants.map_ok(|tenant_shard_id| {
+        let target_ref = &target;
+        let remote_client_ref = &remote_client;
+        async move {
+            let summaries_from_manifests = match gc_tenant_manifests(
```

Comment:
In follow-up work I'd like to use tenant manifest data to check invariants, like e.g. that timelines that are offloaded are also marked as archived. That needs timelines to access tenant specific data.

Also I'm not sure how to cheaply do the `tee` operation: caching all to RAM once is not an option, and cloning the stream (if that's even possible) would, from what I can tell, do the listing operations twice, so it would mean more network operations.

Reply:
Hm, fair points.

But still, this GcSummaryOrContent constructs reads very awkwardly.

Maybe the issue is that we're not constructing the entire stream pipeline in one go, but have this chunk here, then define `gc_timeline` below, then continue consuming the stream.

Might make sense to move `gc_timeline` out of this function body, then use sth like `flatten` or `fuse`?

Reply:
We already do try_flatten below, but moving gc_timeline out of the function body is a sensible change. It bloats the diff a little (mixing code changes that only move stuff with code changes that modify functionality) but I think it's still an okay deal overall given the function is small.

Reply:
done the gc_timeline move